java/algorithm/src/test/java/org/apache/arrow/algorithm/search/TestParallelSearcher.java

```diff
@@ -21,78 +21,71 @@
 
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.stream.Stream;
 
 import org.apache.arrow.algorithm.sort.DefaultVectorComparators;
 import org.apache.arrow.algorithm.sort.VectorValueComparator;
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.RootAllocator;
 import org.apache.arrow.vector.IntVector;
 import org.apache.arrow.vector.VarCharVector;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
  * Test cases for {@link ParallelSearcher}.
  */
-@RunWith(Parameterized.class)
 public class TestParallelSearcher {
 
-  private enum ComparatorType {
+  public enum ComparatorType {
```

Comment:
does this need to be Public?

Reply:
I saw a warning message like this: `Class 'ComparatorType' is exposed outside its defined visibility scope` from the IDE. We can actually ignore this message.

I rolled back for consistency.